src/traces/parcats/calc.js

```diff
@@ -456,16 +456,23 @@ function getUniqueInfo(values, uniqueValues) {
 
 /**
  * Validate the requested display order for the dimensions.
- * If the display order is a permutation of 0 through dimensions.length - 1 then leave it alone. Otherwise, repalce
- * the display order with the dimension order
+ * If the display order is a permutation of 0 through dimensions.length - 1, link to _displayindex
+ * Otherwise, replace the display order with the dimension order
  * @param {Object} trace
  */
 function validateDimensionDisplayInds(trace) {
-    var displayInds = trace.dimensions.filter(visible).map(function(dim) {return dim.displayindex;});
-    if(!isRangePermutation(displayInds)) {
-        trace.dimensions.filter(visible).forEach(function(dim, i) {
-            dim.displayindex = i;
-        });
+    var visibleDims = Lib.filterVisible(trace.dimensions);
```

Comment:
ha nice - I thought about mentioning that one before, do you want to use it to replace the other `filter(visible)` instances here? the `!('visible' in dimension)` clause is likely another relic of @jonmmease's education into `supplyDefaults` 😁

Reply:
Ok - done in https://github.com/plotly/plotly.js/pull/3072/commits/24af02a7ded0d15fe750e8876b049c23adb66b60